### PR TITLE
Added Amazon spider

### PR DESCRIPTION
**Pull request by Darknez07**
Getting the city was a bit tricky so there can be a city in the extras too all and whole, lat and longitudes are good
On the issue of #1613 This pull request stands

### CODE REVIEW

**Comment by Darknez07** [comment]
@iandees Please look into it.

File: locations/spiders/Amazons.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                some = add.split(' ')
+                for i in range(len(some)):
+                    if ',' in some[i] and len(some[i-1])<4 and i >1:
+                        item["extras"] = some[i-1] + " "+some[i]
+                if some[0][0].isdigit():
+                    item["housenumber"] = some[0]
+                    item["street"] = some[1] + " "+some[2]
+                if len(div) == 2:
+                    try:
+                        print()
```
Comment (iandees) [inline locations/spiders/Amazons.py:115]: Please remove prints

File: locations/spiders/Amazons.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+                    item["lat"] = None
+                    item["lon"] = None
+                try:
+                    item['phone']
+                except Exception:
+                    item['phone'] = None
+                if openhours:
+                    item["opening_hours"] = openhours
+                try:
+                    del item["Timings"]
```
Comment (iandees) [inline locations/spiders/Amazons.py:138]: Use `item.pop("Timings", None)` instead of try/except around del.

File: locations/spiders/Amazons.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
+import scrapy
+from scrapy.linkextractors import LinkExtractor
+from scrapy.spiders import CrawlSpider, Rule
+import  requests
+import random
+from locations.items import  GeojsonPointItem
+
+User_Agents = ["Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/44.0.2403.157 Safari/537.36",
```
Comment (iandees) [inline locations/spiders/Amazons.py:8]: I'm not very excited about this. Did the spider fail completely without multiple user agents?

Comment (Darknez07) [reply locations/spiders/Amazons.py:8]: Actually due to lot of requests while testing the spider kept failing, so user agents moreover helps the spider

**Comment by Darknez07** [comment]
@iandees how about now, btw for user-agents it's good to have one during scraping. I been told so.

**Comment by Darknez07** [comment]
@iandees  how about now

**Comment by iandees** [comment]
When you yield items they must have a `ref` value otherwise the system won't work. Please add that.

**Comment by Darknez07** [comment]
@iandees  ok?

> When you yield items they must have a `ref` value otherwise the system won't work. Please add that.

Thanks for this tip

**Comment by Darknez07** [comment]
@iandees  would you like to take a look

**Comment by Darknez07** [comment]
@iandees is it ok now, would you like to merge it now

File: locations/spiders/Amazons.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+User_Agent = "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/44.0.2403.157 Safari/537.36"
+
+
+days = ["Monday","Tuesday","Wednesday","Thursday","Friday","Saturday","Sunday"]
+days_short =["Mon","Tue","Wed","Thur","Fri","Sat","Sun"]
+days_want = ["Mo","Tu","We","Th","Fr","Sa","Su"]
+days_dict = {i:n for n,i in enumerate(days_want)}
+days_rev_dict = {n:i for n,i in enumerate(days_want)}
+class AmazonsSpider(CrawlSpider):
+    name = 'Amazons'
```
Comment (iandees) [inline locations/spiders/Amazons.py:16]: Please use a lower case name.

```suggestion
    name = 'amazon_retail'
```

File: locations/spiders/Amazons.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    allowed_domains = ['amazon.com']
+    # start_urls = ['']
```
Comment (iandees) [inline locations/spiders/Amazons.py:18]: Please don't leave commented-out code.

File: locations/spiders/Amazons.py [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    user_agent = (
```
Comment (iandees) [inline locations/spiders/Amazons.py:19]: I don't think this does anything. Can you look at other spiders to make sure you're setting the user-agent header correctly?

File: locations/spiders/Amazons.py [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+    def set_user_agent(self, request):
+        request.headers["User-Agent"] = self.user_agent
+        return request
+
+    def parse_item(self, response):
+        item = {}
+        openhours = []
+        linker = []
+        locations = None
+        phela = None
```
Comment (iandees) [inline locations/spiders/Amazons.py:44]: This variable is not used anywhere.

File: locations/spiders/Amazons.py [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+        naam = response.xpath("//h2/text()").get()
```
Comment (iandees) [inline locations/spiders/Amazons.py:45]: Please use English variable names.
```suggestion
        name = response.xpath("//h2/text()").get()
```

File: locations/spiders/Amazons.py [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+                                        openhours.append({"Day":days_rev_dict[i],"open":time[2],"close":time[3]})
+                                else:
+                                    for j in range(two,one+1):
+                                        openhours.append({"Day":days_rev_dict[i],"open":time[2],"close":time[3]})
+                            else:
+                                openhours.append({"Day":days[0],"open":time[2],"close":time[3]})
+                    except Exception:
+                        continue
+                    if not (j in [':',': '] and j in ':\n') and len(j) < 20:
+                        ok = j.strip(':').strip('\n')
```
Comment (iandees) [inline locations/spiders/Amazons.py:87]: Please use a better variable name.

File: locations/spiders/Amazons.py [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+                try:
+                    item['phone']
+                except Exception:
+                    item['phone'] = None
+                if openhours:
+                    item["opening_hours"] = openhours
+                item.pop("Timings",None)
+                break
+        if item and len(item) > 2:
+            yield GeojsonPointItem(**item)
```
Comment (iandees) [inline locations/spiders/Amazons.py:130]: You aren't setting a `ref` value on GeojsonPointItem.

Comment (Darknez07) [reply locations/spiders/Amazons.py]: Ohh ok done

**Pull request merged**